modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/PagesWriteSpeedBasedThrottle.java

```diff
@@ -228,7 +230,12 @@ public PagesWriteSpeedBasedThrottle(
             doPark(throttleParkTimeNs);
         }
 
-        pageMemory.metrics().addThrottlingTime(U.nanosToMillis(System.nanoTime() - curNanoTime));
+        long duration = System.nanoTime() - curNanoTime;
+
+        pageMemory.metrics().addThrottlingTime(U.nanosToMillis(duration));
+
+        if (pageMemory.performanceStatistics() != null && pageMemory.performanceStatistics().enabled())
```

Comment:
Can we cache the value of `PerformanceStatisticProcessor` in the instance variable?

Reply:
Can we move this to `addThrottlingTime` inside page memory metrics?

Reply:
PerformanceStatisticProcessor on initializing may be null

moved to addThrottlingTime